tests/test_frame.py

```diff
@@ -368,6 +368,33 @@ def test_types_dropna() -> None:
     res3: None = df.dropna(axis=0, how="all", subset=["col1"], inplace=True)
 
 
+@pytest.mark.parametrize(
+    "drop_arg",
+    [
+        {"AAA"},              # set
+        ["AAA"],              # list
+        ("AAA",),             # tuple
+        {"AAA": None},        # dict
+        "AAA",                # str
+    ]
+)
+def test_types_drop_duplicates(drop_arg) -> None:
```

Comment:
For the purposes of testing type checks, you have to explicitly put in each of these arguments and can't parameterize.   So you'd do:
```python
check(assert_type(df.drop_duplicates({"AAA"}), pd.DataFrame), pd.DataFrame)
check(assert_type(df.drop_duplicates(["AAA"]), pd.DataFrame), pd.DataFrame)
# and similar for the other 3
```
Then you don't need to do `assert_frame_equal()`